src/Microsoft.Security.Utilities.Core/SecretMasker.cs

```diff
@@ -173,175 +176,124 @@ public string MaskSecrets(string input, Action<Detection>? detectionAction = nul
     /// <summary>
     /// Gets or sets the minimum allowable size of a string that's a candidate for masking.
     /// </summary>
-    public virtual int MinimumSecretLength { get; set; }
-
-    /// <summary>
-    /// This implementation assumes no more than one thread is adding regexes, values, or encoders at any given time.
-    /// </summary>
-    [ExcludeFromCodeCoverage]
-    public void AddValue(string value)
+    public int MinimumSecretLength
     {
-        // Test for empty.
-        if (string.IsNullOrEmpty(value))
+        get => _minimumSecretLength;
+        set
         {
-            return;
-        }
-
-        if (value.Length < MinimumSecretLength)
-        {
-            return;
-        }
-
-        var secretLiterals = new List<SecretLiteral>(new[] { new SecretLiteral(value) });
-
-        // Read section.
-        LiteralEncoder[] literalEncoders;
-        try
-        {
-            SyncObject.EnterReadLock();
-
-            // Test whether already added.
-            if (ExplicitlyAddedSecretLiterals.Contains(secretLiterals[0]))
+            SyncObject.EnterWriteLock();
+            try
             {
-                return;
+                _minimumSecretLength = value;
             }
-
-            // Read the value encoders.
-            literalEncoders = LiteralEncoders.ToArray();
-        }
-        finally
-        {
-            if (SyncObject.IsReadLockHeld)
+            finally
             {
-                SyncObject.ExitReadLock();
+                SyncObject.ExitWriteLock();
             }
         }
+    }
+    private int _minimumSecretLength;
 
-        // Compute the encoded values.
-        foreach (LiteralEncoder literalEncoder in literalEncoders)
+    public void AddValue(string value)
+    {
+        if (string.IsNullOrEmpty(value))
```

Comment:
>IsNullOrEmpty [](http://example.com/codeflow?start=19&length=13)

realistically, IsNullOrWhitespace would make sense, no secret literal can comprise all whitespace. but also, why would you pass this data?

Reply:
There's a real enough difference between these that I'm not inclined to change this behavior right now in this PR.

Masking "   " is weird but well-defined.